### PR TITLE
Remove invalid chunk validation

### LINKED ISSUE #2980: [bigtable] InvalidChunk
1. OS type and version
Ubuntu 14.04

2. Python version and virtual environment information `python --version`
Python 2.7.6

3. google-cloud-python version `pip show google-cloud`, `pip show google-<service>` or `pip freeze`

Name: google-cloud-bigtable
Version: 0.22.0

4. Stacktrace if available
```
Traceback (most recent call last):
  File "bt.py", line 13, in <module>
    row = bt_table.read_row('mykey')
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/table.py", line 234, in read_row
    rows_data.consume_all()
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/row_data.py", line 323, in consume_all
    self.consume_next()
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/row_data.py", line 275, in consume_next
    self._validate_chunk(chunk)
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/row_data.py", line 390, in _validate_chunk
    self._validate_chunk_row_in_progress(chunk)
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/row_data.py", line 370, in _validate_chunk_row_in_progress
    _raise_if(not chunk.timestamp_micros or not chunk.value)
  File "/home/jungho.ahn/tmp/test/local/lib/python2.7/site-packages/google/cloud/bigtable/row_data.py", line 441, in _raise_if
    raise InvalidChunk(*args)
google.cloud.bigtable.row_data.InvalidChunk
```
5. Steps to reproduce
See the code example
6. Code example
```
from google.cloud import bigtable

bt_client = bigtable.client.Client(project='myproject', read_only=True)
bt_instance = bt_client.instance(instance_id='myinstance')
bt_table = bt_instance.table(table_id='mytable')

row = bt_table.read_row('mykey')
print row.cells
```

This exception is raised all the time. Did I use incorrectly?

Thanks,

### PR DESCRIPTION
**Pull request by garye**
Fixes #2980

### CODE REVIEW

**Comment by lukesneeringer** [comment]
@dhermes Can you glance at this one? I lack sufficient context.

**Comment by dhermes** [comment]
@garye I'd guess there would be broken tests, but it's hard to say right now because there is a bad release of [`protobuf`][1] that is breaking everything

[1]: https://pypi.python.org/pypi/protobuf/3.2.0

File: bigtable/google/cloud/bigtable/row_data.py [hunk 1]
```diff
@@ -366,8 +366,6 @@ def _validate_chunk_row_in_progress(self, chunk):
         """Helper for :meth:`_validate_chunk`"""
         assert self.state == self.ROW_IN_PROGRESS
         self._validate_chunk_status(chunk)
-        if not chunk.HasField('commit_row') and not chunk.reset_row:
-            _raise_if(not chunk.timestamp_micros or not chunk.value)
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/row_data.py]: Hmm, I thought the timestamp requirement was still good, just not the requirement for a non-empty value.

Comment (garye) [inline bigtable/google/cloud/bigtable/row_data.py]: timestamp == 0 is valid, so if not chunk.timestamp_micros is true in this case then we should remove it.

Comment (dhermes) [inline bigtable/google/cloud/bigtable/row_data.py]: The `protobuf` types don't use `None` as "emtpy", they just use the default values for their type, so setting `timestamp_micros = 0` is the same as not setting it at all

Comment (tseaver) [inline bigtable/google/cloud/bigtable/row_data.py]: > setting timestamp_micros = 0 is the same as not setting it at all

Right:  the question is, is it valid to be start a new cell without a timestamp? (we are in the "row-in-progress" state, which means we have no active / partial cell).  I'm fine to defer to @garye if the answer is "yes, that is valid."

It'd be breat to have public docs for the chunking stuff, too. ;)

Comment (garye) [inline bigtable/google/cloud/bigtable/row_data.py]: Yep, would be great to have docs on this :) Timestamp of 0 is indeed valid.

**Comment by lukesneeringer** [comment]
@dhermes Update on the `protobuf` thing: googleapis/api-client-staging#159

**Comment by dhermes** [comment]
I restarted the CircleCI build: https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/865

Hopefully it works?

**Comment by garye** [comment]
Can someone help me finish this off? The test failures don't really seem related. Getting this fix pushed and released will unblock a number of users.

**Comment by dhermes** [comment]
@garye Just re-started the CircleCI build: https://circleci.com/gh/GoogleCloudPlatform/google-cloud-python/876

Hopefully the bad imports are fully resolved. I'll follow-up if not.

**Comment by dhermes** [comment]
@garye So we've got a unit test failing that is actually caused by this PR:

```

    def test_invalid_empty_second_chunk(self):
        from google.cloud.bigtable.row_data import InvalidChunk
    
        chunks = _generate_cell_chunks(['', ''])
        first = chunks[0]
        first.row_key = b'RK'
        first.family_name.value = 'A'
        first.qualifier.value = b'C'
        response = _ReadRowsResponseV2(chunks)
        iterator = _MockCancellableIterator(response)
        prd = self._make_one(iterator)
        with self.assertRaises(InvalidChunk):
>           prd.consume_next()
E           AssertionError: InvalidChunk not raised
```

ISTM the right move is just to delete this test case since it no longer applies?

**Comment by garye** [comment]
yeah, i think that's an invalid test case

**Comment by dhermes** [comment]
Ping me when this PR is updated with the test case removed? You can ignore the speech test failures. It looks like the underlying protobuf libraries are (now correctly) using single instead of double precision for message fields with single precision

**Comment by garye** [comment]
Will do, thanks

File: bigtable/unit_tests/test_row_data.py [hunk 2]
```diff
@@ -416,21 +416,6 @@ def test_invalid_empty_chunk(self):
...
-        first.row_key = b'RK'
-        first.family_name.value = 'A'
-        first.qualifier.value = b'C'
-        response = _ReadRowsResponseV2(chunks)
-        iterator = _MockCancellableIterator(response)
-        prd = self._make_one(iterator)
-        with self.assertRaises(InvalidChunk):
-            prd.consume_next()
-
-
```
Comment (tseaver) [inline bigtable/unit_tests/test_row_data.py]: @garye pylint is complaining:  you removed one too-many blank lines there at the end of the class. :(

Comment (tseaver) [inline bigtable/unit_tests/test_row_data.py]:
```python
bigtable/unit_tests/test_row_data.py:419:1: E302 expected 2 blank lines, found 1
```

**Comment by garye** [comment]
Yeah, thanks I noticed that and (hopefully) pushed a fix.

**Comment by tseaver** [comment]
@garye Ugh, you need to be able to run `tox -e lint` locally:

```python
bigtable/unit_tests/test_row_data.py:419:1: W293 blank line contains whitespace
```

**Comment by garye** [comment]
Ugh, OK. That will take a bit of time to set up.

**Comment by tseaver** [comment]
I *think* it should just work to run `tox -e pylint`.  You probably want to export the following environment variables so it only runs on your changes:

```bash
$ export GOOGLE_CLOUD_TESTING_REMOTE=upstream  # the name of the remote for this repo
$ export GOOGLE_CLOUD_TESTING_BRANCH=master
$ tox -e lint
```

**Comment by dhermes** [comment]
LGTM

**Comment by garye** [comment]
Thanks guys!

**Comment by dhermes** [comment]
@garye We hope to cut a release in the next 2-3 days as well

**Pull request merged**